src/hwdb/hwdb.c

```diff
@@ -422,20 +419,13 @@ static int trie_store(struct trie *trie, const char *filename) {
         /* write header */
         size = ftello(t.f);
         h.file_size = htole64(size);
-        err = fseeko(t.f, 0, SEEK_SET);
-        if (err < 0) {
-                fclose(t.f);
-                unlink_noerrno(filename_tmp);
-                return -errno;
-        }
+        if (fseeko(t.f, 0, SEEK_SET) < 0)
+                goto error;
```

Comment:
This will leave `r` with a wrong value -- shouldn't that be updated to the result of this `fseeko()`?